tests/test_single_topology.py

```diff
@@ -175,9 +170,29 @@ def testing_find_dummy_groups_and_multiple_anchors():
     core_b = [2, 1, 4, 3]
 
     with pytest.warns(MultipleAnchorWarning):
-        dgs, jks = single_topology.find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b)
-        assert dgs == [{0}]
-        assert jks == [(1, 2)]
+        dgs = single_topology.find_dummy_groups_and_anchors(mol_a, mol_b, core_a, core_b)
+        assert dgs == {1: (2, {0})}
+
+
+def test_ethane_cyclobutane():
+    """Test case where a naive heuristic for identifying dummy groups results in disconnected components"""
+
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("CC"))
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("c1ccc1"))
+
+    AllChem.EmbedMolecule(mol_a, randomSeed=2022)
+    AllChem.EmbedMolecule(mol_b, randomSeed=2022)
+
+    core = np.array([[2, 0], [4, 2], [0, 3], [3, 7]])
+    ff = Forcefield.load_default()
+    st = SingleTopology(mol_a, mol_b, core, ff)
+
+    g = nx.Graph()
+    g.add_nodes_from(range(st.get_num_atoms()))
+    g.add_edges_from(st.src_system.bond.potential.idxs)
+
+    # bond graph should be connected (i.e. no floating bits)
+    assert len(list(nx.connected_components(g))) == 1
```

Comment:
Should this check just be added to the SingleTopology init?

Reply:
Good call, added this in https://github.com/proteneer/timemachine/pull/1019/commits/afbf684951fb5af969ceaaeb399a69d03344760d